### PR TITLE
Add save shortcut for navigation menu screen

### PR DESCRIPTION
**Pull request by talldan**
## Description
Related #21338

Adds a save shortcut for the navigation menu screen.

It does this via a `MenuEditorShortcuts` component that might be useful for registering any other shortcuts.

## How has this been tested?
1. Edit a menu on the Navigation Menu page.
2. Press Cmd+S
3. Reload the page and observe that change is present

Note - when saving there's no success/error notifications. I've created #21344 to track that.

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist:
- [ ] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Review by youknowriad — APPROVED** [review]
Code wise, this looks good (I didn't test).
It would be good to extract at some point the keyboard shortcuts modal to the keyboard shortcuts package in order to reuse in multiple screen. It is still 100% generated dynamically yet though.

File: packages/edit-navigation/src/components/menu-editor/shortcuts.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+function MenuEditorShortcuts( { saveBlocks } ) {
+	useShortcut(
+		'core/edit-navigation/save-menu',
+		useCallback( ( event ) => {
+			event.preventDefault();
+			saveBlocks();
+		} ),
+		{
+			bindGlobal: true,
+			preventDefault: true,
```
Comment (youknowriad) [inline packages/edit-navigation/src/components/menu-editor/shortcuts.js:18]: I don't think this option exists?

File: packages/edit-navigation/src/components/menu-editor/shortcuts.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+				modifier: 'primary',
+				character: 's',
+			},
+		} );
+	}, [ registerShortcut ] );
+
+	return null;
+}
+
+MenuEditorShortcuts.Register = RegisterMenuEditorShortcuts;
```
Comment (youknowriad) [inline packages/edit-navigation/src/components/menu-editor/shortcuts.js:42]: I know we used this pattern in the other packages too, but I wonder if a hook is better than a component for the registration :) (not specific to this PR though)

File: packages/edit-navigation/src/components/menu-editor/shortcuts.js (hunk 1 above)
Comment (talldan) [reply packages/edit-navigation/src/components/menu-editor/shortcuts.js:18]: Oh yep, forgot to remove it 😄

**Review by draganescu — COMMENTED** [review]
Tested and worked as described.

**Comment by talldan** [comment]
I need to update the package-lock file by the looks of it. Will do that tomorrow and then merge. Thanks for the speedy reviews!

**Pull request merged**